hw/top_earlgrey/sw/autogen/devicetables.c

```diff
@@ -0,0 +1,2181 @@
+        },
+        [kDtSysrstCtrlPinPwrbIn] = {
+          .__internal = {
+            .type = kDtPinTypeMio,
+            .periph_input_or_direct_pad = kTopEarlgreyPinmuxPeripheralInSysrstCtrlAonPwrbIn,
+            .outsel = kDtPinOutselNone,
+          }
+        },
+        [kDtSysrstCtrlPinLidOpen] = {
+          .__internal = {
+            .type = kDtPinTypeMio,
+            .periph_input_or_direct_pad = kTopEarlgreyPinmuxPeripheralInSysrstCtrlAonLidOpen,
+            .outsel = kDtPinOutselNone,
+          }
+        },
+        [kDtSysrstCtrlPinBatDisable] = {
+          .__internal = {
+            .type = kDtPinTypeMio,
+            .periph_input_or_direct_pad = kDtPinPeriphInputNone,
+            .outsel = kTopEarlgreyPinmuxOutselSysrstCtrlAonBatDisable,
+          }
+        },
+        [kDtSysrstCtrlPinKey0Out] = {
+          .__internal = {
+            .type = kDtPinTypeMio,
+            .periph_input_or_direct_pad = kDtPinPeriphInputNone,
+            .outsel = kTopEarlgreyPinmuxOutselSysrstCtrlAonKey0Out,
+          }
+        },
+        [kDtSysrstCtrlPinKey1Out] = {
+          .__internal = {
+            .type = kDtPinTypeMio,
+            .periph_input_or_direct_pad = kDtPinPeriphInputNone,
+            .outsel = kTopEarlgreyPinmuxOutselSysrstCtrlAonKey1Out,
+          }
+        },
+        [kDtSysrstCtrlPinKey2Out] = {
+          .__internal = {
+            .type = kDtPinTypeMio,
+            .periph_input_or_direct_pad = kDtPinPeriphInputNone,
+            .outsel = kTopEarlgreyPinmuxOutselSysrstCtrlAonKey2Out,
+          }
+        },
+        [kDtSysrstCtrlPinPwrbOut] = {
+          .__internal = {
+            .type = kDtPinTypeMio,
+            .periph_input_or_direct_pad = kDtPinPeriphInputNone,
+            .outsel = kTopEarlgreyPinmuxOutselSysrstCtrlAonPwrbOut,
+          }
+        },
+        [kDtSysrstCtrlPinZ3Wakeup] = {
+          .__internal = {
+            .type = kDtPinTypeMio,
+            .periph_input_or_direct_pad = kDtPinPeriphInputNone,
+            .outsel = kTopEarlgreyPinmuxOutselSysrstCtrlAonZ3Wakeup,
+          }
+        },
+        [kDtSysrstCtrlPinEcRstL] = {
+          .__internal = {
+            .type = kDtPinTypeDio,
+            .periph_input_or_direct_pad = kTopEarlgreyDirectPadsSysrstCtrlAonEcRstL,
+            .outsel = 0,
+          }
+        },
+        [kDtSysrstCtrlPinFlashWpL] = {
+          .__internal = {
+            .type = kDtPinTypeDio,
+            .periph_input_or_direct_pad = kTopEarlgreyDirectPadsSysrstCtrlAonFlashWpL,
+            .outsel = 0,
+          }
+        },
+      },
+    },
+  },
+};
+// Device tables for uart
+_Static_assert(kDtUartRegBlockCount == 1, "Reg block count mismatch");
+_Static_assert(kDtUartClockCount == 1, "Clock count mismatch");
+_Static_assert(kDtUartIrqTypeCount == 9, "IRQ count mismatch");
+_Static_assert(kDtUartCount == 4, "Number of uart modules mismatch");
+
+const dt_uart_t kDtUart[kDtUartCount] = {
+  // Properties for uart0
```

Comment:
```suggestion
[kDtUart0] = 
```

Reply:
At the moment, there is no enumeration for the indices of each block, ie there is no `kDtUart0`, there is just `kDtUart[0]`. Maybe we could create such an enumeration with per-top names but I don't know if that's really useful? The only instance where it could potentially be useful is if we had a top with two instances of a block and those instances are not obvious `0` and `1` but something more meaningful.